### ISSUE TITLE
curl 7.73.0 for Windows upload performance is extremely bad when server only supports 64KB TCP windows.

### ISSUE THREAD

**Issue by tpodom** [curl/curl#6276]
### I did this
Uploaded a file using curl 7.73.0_05 for Windows to a node.js Windows server that accepts a simple multipart/form-data POST. In this example, I uploaded the curl-7.72.0.zip file because it is several megabytes in size.

Using curl 7.73.0, the request takes 45 seconds and then times out.

Using curl 7.72.0, the request takes 1.09 seconds.

Command Line issued:
curl.exe -H "Content-Type: multipart/form-data" -F "data=@curl-7.72.0.zip" -w  "%{time_total}\n" http://server:8989/upload

I am unable to reproduce this using docker curlimages/curl:7.73.0.

To reproduce this, I ran "Set-NetTCPSetting -AutoTuningLevelLocal Disabled" on the server before starting the upload-test-server. 

I have an example test server at https://github.com/tpodom/upload-test-server which includes packet captures [curl_7_72.pcapng](https://github.com/tpodom/upload-test-server/blob/main/curl_7_72.pcapng) and [curl_7_73.pcapng](https://github.com/tpodom/upload-test-server/blob/main/curl_7_73.pcapng).  As you can see in the 7.73 packet captures, there are some significant delays on the client before it resumes sending after receiving ACKs from the server.

Even when the server is set to AutoTuningLevelLocal Normal to support larger TCP window sizes, the performance of curl 7.73 on Windows is noticeably slower.  With larger window sizes, curl 7.73 took 8 seconds to upload which was a lot faster than the original issue but curl 7.72 still only takes just over 1 second.

### I expected the following
curl 7.73.0 to be as fast as 7.72.0 or at least close in performance.

### curl/libcurl version
curl 7.73.0 (x86_64-pc-win32) libcurl/7.73.0 OpenSSL/1.1.1h (Schannel) zlib/1.2.11 brotli/1.0.9 zstd/1.4.5 WinIDN libssh2/1.9.0 nghttp2/1.42.0
Release-Date: 2020-10-14
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps mqtt pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: AsynchDNS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile MultiSSL NTLM SPNEGO SSL SSPI TLS-SRP Unicode UnixSockets brotli libz zstd

curl 7.72.0 (x86_64-pc-win32) libcurl/7.72.0 OpenSSL/1.1.1g (Schannel) zlib/1.2.11 brotli/1.0.7 zstd/1.4.5 WinIDN libssh2/1.9.0 nghttp2/1.41.0
Release-Date: 2020-08-19
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps mqtt pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: AsynchDNS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile MultiSSL NTLM SPNEGO SSL SSPI TLS-SRP Unicode UnixSockets brotli libz zstd

### operating system

Client: Windows 10 Enterprise build 2004
Server: Windows Server 2019 Standard build 1809

**Comment by tpodom** [comment]
I ran through the git bisect from curl-7_72_0 to curl-7_73_0 and this was the resulting commit that affected the performance:

git bisect bad
d2a7d7c185f98df8f3e585e5620cbc0482e45fac is the first bad commit

**Comment by bagder** [comment]
Duplicate of #6146
